src/urllib3/poolmanager.py

```diff
@@ -342,8 +342,12 @@ def urlopen(self, method, url, redirect=True, **kw):
         # conn.is_same_host() which may use socket.gethostbyname() in the future.
         if (retries.remove_headers_on_redirect
                 and not conn.is_same_host(redirect_location)):
-            for header in retries.remove_headers_on_redirect:
```

Comment:
Let's keep some of the original structure of this logic by instead looping through all the header names with `six.viewkeys()` (so we don't have to make a copy) and then seeing if that header name is within the `remove_headers_on_redirect` frozenset.

Reply:
I changed by 4357014. But made a copy by `.copy()` due to getting `RuntimeError: dictionary changed size during iteration` if without `.copy()`.